src/domain/repositories/ProgramsRepository.ts

```diff
@@ -1,20 +1,47 @@
-import { Async } from "domain/entities/Async";
-import { Id } from "domain/entities/Base";
-import { Pair } from "scripts/common";
+import { Enrollment, Event, TrackedEntity } from "domain/entities/TrackedEntity";
+import { CancelableResponse } from "types/d2-api";
 
 export interface ProgramsRepository {
-    closePatients(options: ClosePatientsOptions): Async<void>;
+    get(options: GetOptions): CancelableResponse<ApiGetResponse>;
+    save(payload: Payload): CancelableResponse<ApiSaveResponse>;
 }
 
-export interface ClosePatientsOptions {
-    orgUnitsIds?: Id[];
+export interface GetOptions {
+    programId: string;
+    orgUnitsIds?: string[];
     startDate?: string;
     endDate?: string;
-    programId: Id;
-    programStagesIds: Id[];
-    closureProgramId: Id;
-    timeOfReference: string;
-    pairsDeValue: Pair[];
-    comments?: Pair;
-    post: boolean;
+}
+
+export interface Payload {
+    enrollments: Enrollment[];
+    events: Event[];
+}
+
+export interface ApiGetResponse {
```

Comment:
As the name shows, it's an API detail implementation, and should not be in the domain (it should be in the data repository, not exported). It's the API that decides the `instance` key, but we can decide a more declarative name, for example: `{trackedEntities: TrackedEntity[]}`.